### PR TITLE
[react-leaflet] Add missing onZoom event.

### PR DESCRIPTION
**Pull request by EliteMasterEric**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://leafletjs.com/reference-1.7.1.html#map-zoom
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

---

Fixes issue #34217. Also adds self as definition owner.

### CODE REVIEW

File: types/react-leaflet/react-leaflet-tests.tsx [hunk 1]
```diff
@@ -1048,4 +1048,4 @@ const mapEvents: MapEvents = {
         console.log(event.target);
         console.log(event.popup);
     },
-}
\ No newline at end of file
+};
```
Comment (fenech) [inline types/react-leaflet/react-leaflet-tests.tsx:1051]: I guess that this can just be squashed into the previous commit?

**Review by fenech — COMMENTED** [review]
Looks good to me! But I have to admit that I'm not very clear on the best testing strategy for this. Maybe one of the other reviewers has some more useful feedback to leave, related to this.

**Comment by MasterEric** [comment]
@fenech Commits have been squashed.

File: types/react-leaflet/react-leaflet-tests.tsx (hunk 1 above)
Comment (MasterEric) [reply types/react-leaflet/react-leaflet-tests.tsx:1051]: Done.

**Comment by MasterEric** [comment]
> But I have to admit that I'm not very clear on the best testing strategy for this. 

Keep in mind the only issue is ensuring types validate properly. The JavaScript is never actually compiled or run, the tests only exist to receive type validation; if it fails, it's likely that existing user code will fail type validation too.

As such, I chose the strategy of creating an event listener for each event, then adding a line that accesses some of the properties of the event they receive. That way, if any listener becomes unable to access the appropriate property, the test will fail.

File: types/react-leaflet/react-leaflet-tests.tsx [hunk 2]
```diff
@@ -887,3 +888,164 @@ export class GeoJSONExample extends Component<undefined, undefined> {
         </Map>);
     }
 }
+
+const mapEvents: MapEvents = {
+    onclick: (event: L.LeafletMouseEvent) => {
+        console.log(event);
+        console.log(event.target);
+        console.log(event.originalEvent);
+    },
```
Comment (davschne) [inline types/react-leaflet/react-leaflet-tests.tsx:897]: This is just an idea, but something like the following could be a more compact way of accomplishing the same thing:
```typescript
onclick: ({ target, originalEvent }: L.LeafletMouseEvent) => { }
```

**Review by davschne — APPROVED** [review]

**Comment by MasterEric** [comment]
Ready to merge

**Pull request merged**